frontend/webapp/reuseable-components/status/index.tsx

```diff
@@ -1,100 +1,103 @@
-  color: ${({ $isActive }) => ($isActive ? '#51DB51' : '#DB5151')};
-  text-transform: ${({ $withSpecialFont }) => ($withSpecialFont ? 'uppercase' : 'unset')};
+const Title = styled(Text)<{
+  $isPale: StatusProps['isPale'];
+  $isActive: StatusProps['isActive'];
+}>`
+  color: ${({ $isPale, $isActive, theme }) => ($isPale ? theme.text.secondary : $isActive ? theme.text.success : theme.text.error)};
 `;
 
-const TextDivider = styled.div<{ $isActive?: Props['isActive'] }>`
-  width: 1px;
-  height: 12px;
-  background: ${({ $isActive }) => ($isActive ? 'rgba(124, 237, 124, 0.16)' : 'rgba(237, 124, 124, 0.16)')};
-  margin: 0 8px;
+const SubTitle = styled(Text)<{
+  $isPale: StatusProps['isPale'];
+  $isActive: StatusProps['isActive'];
+}>`
+  color: ${({ $isPale, $isActive }) => ($isPale ? theme.text.grey : $isActive ? '#51DB51' : '#DB5151')};
 `;
 
-const Status: React.FC<Props> = ({ title, subtitle, isActive, withIcon, withBorder, withBackground, withSpecialFont, withSmaller }) => {
+export const Status: React.FC<StatusProps> = ({ title, subtitle, size = 12, family = 'secondary', isPale, isActive, withIcon, withBorder, withBackground }) => {
   return (
-    <StatusWrapper $isActive={isActive} $withIcon={withIcon} $withBorder={withBorder} $withBackground={withBackground} $withSmaller={withSmaller}>
+    <StatusWrapper $size={size} $isPale={isPale} $isActive={isActive} $withIcon={withIcon} $withBorder={withBorder} $withBackground={withBackground}>
       {withIcon && (
-        <IconWrapper $withSmaller={withSmaller}>
-          <Image src={getStatusIcon(isActive ? 'success' : 'error')} alt='status' width={withSmaller ? 12 : 16} height={withSmaller ? 14 : 16} />
+        <IconWrapper>
+          {/* TODO: SVG to JSX */}
+          <Image src={isPale ? `/icons/common/circled-${isActive ? 'check' : 'cross'}.svg` : getStatusIcon(isActive ? 'success' : 'error')} alt='status' width={size + 2} height={size + 2} />
         </IconWrapper>
       )}
 
-      <TextWrapper>
-        <Title $isActive={isActive} $withSpecialFont={withSpecialFont} $withSmaller={withSmaller}>
-          {title || (isActive ? 'Active' : 'Inactive')}
-        </Title>
+      {(!!title || !!subtitle) && (
+        <TextWrapper>
+          {!!title && (
+            <Title size={size} family={family} $isPale={isPale} $isActive={isActive}>
+              {title}
+            </Title>
+          )}
 
-        {subtitle && (
-          <TextWrapper>
-            <TextDivider $isActive={isActive} />
-            <SubTitle $isActive={isActive} $withSpecialFont={withSpecialFont} $withSmaller={withSmaller}>
-              {subtitle}
-            </SubTitle>
-          </TextWrapper>
-        )}
-      </TextWrapper>
+          {!!subtitle && (
+            <TextWrapper>
+              <Divider orientation='vertical' length={`${size - 2}px`} color={isPale ? undefined : (isActive ? theme.text.success : theme.text.error) + hexPercentValues['030']} />
```

Comment:
handle the style inside the styled component block to reduce the theme path's dependency.